documentation/general/torn-sdk.md

```diff
@@ -0,0 +1,109 @@
+| Visual Studio Design Time | From Visual Studio | From Both |
+
+Generally this mixing of components is fine because Visual Studio will install a .NET SDK that is functionally paired with it. That is the compiler and analyzers are the same hence mixing poses no real issue. For example 17.10 installs .NET SDK 8.0.3xx, 17.9 installs .NET SDK 8.0.2xx, etc ... However when the .NET SDK is not paired with the Visual Studio version,then compatibility issues can, and will, arise. Particularly when the .NET SDK is _newer_ than Visual Studio customers will end up with the following style of error:
+
+> CSC : warning CS9057: The analyzer assembly '..\dotnet\sdk\8.0.200\Sdks\Microsoft.NET.Sdk.Razor\source-
+generators\Microsoft.CodeAnalysis.Razor.Compiler.SourceGenerators.dll' references version '4.9.0.0' of the compiler, which is newer
+than the currently running version '4.8.0.0'.
+
+To address this issue we are going to separate Visual Studio and the .NET SDK in build scenarios. That will change our matrix to the following:
+
+| Scenario | Loads Roslyn | Loads Analyzers / Generators |
+| --- | --- | --- |
+| msbuild | From .NET SDK | From .NET SDK |
+| dotnet msbuild | From .NET SDK | From .NET SDK |
+| Visual Studio Design Time | From Visual Studio | From Visual Studio |
+
+Specifically we will be
+
+1. Changing msbuild to use a compiler from the .NET SDK
+2. Changing Visual Studio to use analyzers from Visual Studio
+
+In addition to making our builds more reliable this will also massively simplify our analyzer development strategy. Analyzers following this model can always target the latest Roslyn version without the need for complicated multi-targeting.
+
+## Detailed Design
+
+### MSBuild using .NET SDK Compiler
+
+The .NET SDK will start producing a package named Microsoft.NETSdk.Compiler.Roslyn. This will contain a .NET Framework version of the Roslyn compiler that matches .NET Core version. This will be published as a part of the .NET SDK release process meaning it's available for all publicly available .NET SDKs.
+
+The .NET SDK has the capability to [detect a torn state][pr-detect-torn-state]. When this is detected the matching Microsoft.NETSDK.Compiler.Roslyn package for this version of the .NET SDK will be downloaded via `<PackageDownload>`. Then the `$(RoslynTargetsPath)` will be reset to point into the package contents. This will cause MSBuild to load the Roslyn compiler from that location vs. the Visual Studio location.
+
+One downside to this approach is that it is possible to end up with two VBCSCompiler server processes. Consider that when a solution has a mix of .NET SDK style projects and legacy projects then in a torn state the .NET SDK projects will use the .NET SDK compiler will legacy projects will use the Visual Studio compiler. While not a desirable outcome, it is a correct one. Customers who wish to only have one VBCSCompiler should correct the torn state in their build tools.
```

Comment:
What does "legacy projects" mean here, like .NET Framework?

Reply:
These are the format of msbuild files before the .NET SDK was created. The easiest way to distinguish between them is the presence / absence of the `Sdk =` attribute in the `<Project>` element. It's a bit more nuanced than that but it's a quick test.

They build using the tasks / targets inside of MSBuild not the .NET SDK.

Reply:
```suggestion
One downside to this approach is that it is possible to end up with two VBCSCompiler server processes. Consider that when a solution has a mix of .NET SDK style projects and non-SDK .NET Framework projects then in a torn state the .NET SDK projects will use the .NET SDK compiler will legacy projects will use the Visual Studio compiler. While not a desirable outcome, it is a correct one. Customers who wish to only have one VBCSCompiler should correct the torn state in their build tools.
```

Reply:
Will add this in.

Reply:
Cleaned up the terminology a bit here too.